src/core/searchUtils.js

```diff
@@ -108,10 +108,22 @@ export function loadByCategoryIfNeeded(
   { store: { dispatch, getState }, location, params }
 ) {
   const state = getState();
+
+  // HACK: This loads Firefox personas (lightweight themes) for Android
+  // until https://github.com/mozilla/addons-frontend/issues/1723#issuecomment-278793546
+  // is more fully addressed.
+  // Essentially: right now there are no categories for the combo
+  // of "Android" + "Themes" but Firefox lightweight themes will work fine
+  // on mobile so we request them.
+  // Obviously we need to fix this on the API end so our requests aren't
+  // overridden, but for now this will work.
+  const clientApp = params.application === 'android' ?
```

Comment:
The `loadByCategoryIfNeeded` function and the `mapStateToProps` function both look like they end up calling `core/api:search()`. Can't you just put the android->firefox hack in `search()` so that it only gets switched in one place?